File: backend/src/main/java/com/khata4u/backend/controller/AuthController.java
package com.khata4u.backend.controller;

import com.khata4u.backend.dto.LoginRequest;
import com.khata4u.backend.dto.LoginResponse;
import com.khata4u.backend.dto.RegisterRequest;
import com.khata4u.backend.service.AuthService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;
    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/register")
    public ResponseEntity<LoginResponse> register(@RequestBody RegisterRequest req) {
        return ResponseEntity.ok(authService.register(req));
    }

    @PostMapping("/login")
    public ResponseEntity<LoginResponse> login(@RequestBody LoginRequest req) {
        return ResponseEntity.ok(authService.login(req));
    }
}

File: backend/src/main/java/com/khata4u/backend/AuthController.kt
package com.khata4u.backend

// NOTE: This file previously exposed a permissive mock /auth endpoint that accepted any
// username/password length >= 6 and issued a JWT without checking the database. The
// real secure endpoints live under /api/auth and are implemented in
// com.khata4u.backend.controller.AuthController (Java). To avoid accidental exposure
// during development this Kotlin controller has been converted into a non-controller helper.

// data class LoginRequest(val username: String, val password: String)
// data class JwtResponse(val token: String)

// Removed @RestController and @RequestMapping annotations so this class won't register endpoints.
class AuthControllerHelper {
    // helper functions could go here if needed in the future
}
